tox.ini

```diff
@@ -0,0 +1,67 @@
+[tox]
+skip_missing_interpreters = True
+skipsdist=True
+minversion = 1.8
+envlist =
+    py27-trollius,
+    py32-trollius,
+    py33-trollius,
+    py33-tulip,
+    py34,
+    py35,
+    pypy-trollius,
+    pypy3-trollius,
+    py2-pep8,
+    py3-pep8,
+    packaging
+
+[testenv]
+# This is required in order to get UTF-8 output inside of the subprocesses
+# that our tests use.
+setenv = LC_CTYPE = en_US.UTF-8
+# Pass Display down to have it for the tests available
+passenv = DISPLAY
+whitelist_externals = /bin/bash
+# Install trollius for 2, 3.2, and pypy
+# Install either trollius or tulip for 3.3
+# Asyncio is in the standard library for 3.4+
+deps =
+    nose-cov
+
+    trollius: trollius
+    tulip: asyncio
+
+    # https://github.com/travis-ci/travis-ci/issues/4866
+    py32: coverage<4.0.0
+commands =
+    # xcffib has to be installed before cairo
+    pip install xcffib
+    # Install unpinned requirements from requirements.in
+    pip install -r {toxinidir}/requirements.in
+    # build pangocffi module
+    bash -c "python -c 'import cffi, sys; sys.exit(cffi.__version_info__[0])' || python {toxinidir}/libqtile/ffi_build.py"
```

Comment:
It's not great to have to do this, especially since tox will install qtile and this should be run then. With cairocffi, a way was found to run from the tox installed package, https://github.com/SimonSapin/cairocffi/blob/master/tox.ini